src/screen/trip.js

```diff
@@ -1,18 +1,91 @@
-import React from "react";
+import React, {useState} from "react";
 import { Fab } from "../Components/common/Fab";
-import { Food } from "../Components/Food";
+import { Food }from "../Components/Food";
+import './trip.css';
+import { MapContainer, TileLayer, Marker, Popup } from 'react-leaflet';
+import Search from "../Search";
+import RoutingMachine from "./routing";
+import ArrowRightAltIcon from '@mui/icons-material/ArrowRightAlt';
+
 
 const navigateToHome = () => (window.location.href = "/");
 
 export function Trip() {
+  const [currentCity, setCurrentCity] = useState(null);
+  const [fromCity, setFromCity] = useState(null);
+  const [toCity, setToCity] = useState(null);
+  const [cityList, setCityList] = useState([]);
+  const [isLoaded, setLoaded] = useState(false);
+
+  function setRoute() {
+    if(fromCity==null || toCity==null) {
+      alert("Something went wrong, try again!");
+      return;
+    }
+    fetch(`https://geocode.search.hereapi.com/v1/geocode?q=${fromCity}&apiKey=${process.env.REACT_APP_HEREAPI}`)
+    .then((res) => res.json())
+      .then((result) => {
+        //console.log(result);
+        let list = [...cityList];
+        list.push({
+          lat: result.items[0].position.lat,
+          lon: result.items[0].position.lng,
+        });
+        setCityList(list);
+      });
+    fetch(`https://geocode.search.hereapi.com/v1/geocode?q=${toCity}&apiKey=${process.env.REACT_APP_HEREAPI}`)
+      .then((res) => res.json())
+        .then((result) => {
+          //console.log(result);
+          let list = [...cityList];
+          list.push({
+            lat: result.items[0].position.lat,
+            lon: result.items[0].position.lng,
+          });
+          setCityList(list);
+          setLoaded(true);
+        });
+      
+  }
+
+
   React.useEffect(() => {
     document.title = "Plan a trip";
-  }, []);
 
-  
+    window.navigator.geolocation.getCurrentPosition((position) => {
+      fetch(`https://api.openweathermap.org/data/2.5/weather?lat=${position.coords.latitude}&lon=${position.coords.longitude}&appid=${process.env.REACT_APP_APIKEY}`)
```

Comment:
Similar code exists in `/screen/home.js`. You can receive coordinates and city as props and save this API call. 

CC: @Krishi-02

Reply:
FYI: you are using `lon` to denote longitude whereas the rest of the app is using `lng`. 

If you can replace your `lon` with `lng`, that would be great. Otherwise, don't forget to change the name of that key.